app/controllers/v1/tasks_controller.rb

```diff
@@ -0,0 +1,116 @@
+module V1
+  class TasksController < ApplicationController
+    swagger_controller :tasks, 'Tasks'
+
+    swagger_api :index do
+      summary 'List all tasks'
+      notes 'This lists all the active tasks'
+      param :path, :project_id, :string, :required, 'Project Id'
+      param :query, :page, :integer, :optional, 'page number of results, default 1'
+      param :query, :page_size, :integer, :optional, 'number of results per page, default 25'
+    end
+    def index
+      tasks, errors = ListTasks.new(index_params).call
+      if errors.any?
+        render json: { errors: errors }, status: 400
+      else
+        render json: tasks
+      end
+    end
+
+    swagger_api :show do
+      summary 'Fetch a single Task'
+      param :path, :project_id, :string, :required, 'Project Id'
+      param :path, :id, :string, :required, 'Task Id'
+    end
+    def show
+      task = Task.find_by id: params[:id]
```

Comment:
Since you nested the routes for Task under Project, it might be good to find by both the task id and the project id since you have them both.

Reply:
Great catch, thank you!